src/eea/themecentre/browser/datareporting.py

```diff
@@ -62,12 +59,22 @@ def relatedReportingObligations(self):
             reso = res.getObject()
             for rodid in reso.reportingObligations:
                 if rodid not in rodsdone:
+
+                    try:
+                        rodinfo = rodsinfo[int(rodid)]
+                    except KeyError, err:
+                        # Obligation not found
+                        logger.exception(err)
+                        continue
+
                     rodsdone.append(rodid)
                     rodurl = rodbaseurl + rodid
+                    roddescription = rodinfo['DESCRIPTION']
+                    rodtitle = rodinfo['TITLE']
                     rods.append({
                         'id': rodid,
-                        'Description': rodsinfo[int(rodid)]['DESCRIPTION'],
-                        'Title': rodsinfo[int(rodid)]['TITLE'],
+                        'Description': roddescription,
```

Comment:
We should get rid of intermediate variable `roddescription` and directly use `rodinfo.get('DESCRIPTION', "")`

Reply:
fixed